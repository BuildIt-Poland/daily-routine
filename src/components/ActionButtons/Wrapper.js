import styled from 'styled-components';

import media from '../../styles/media';
import { colorWhite } from '../../styles/designTokens';
import Section from '../Section';

const Wrapper = styled(Section)`
  background: ${colorWhite};

  ${media.phone`
    background: inherit;
  `};
`;

export default Wrapper;
