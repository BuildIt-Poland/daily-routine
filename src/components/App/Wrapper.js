import React, { useEffect, useState } from 'react';
import styled, { withTheme } from 'styled-components';
import { animated, useTransition, config } from 'react-spring';
import { colorLightTeal } from '../../styles/designTokens';
import extractRoleFromPath from '../../utils/extractRoleFromPath';
import generateColors from '../../utils/generateColors';

const Wrapper = styled(animated.div)`
  position: absolute;
  width: 100%;
  min-height: 100%;
  display: grid;
  grid-template-rows: auto 1fr auto;
`;

function Animator({ children, location, theme }) {
  const { pathname } = location;
  const { primaryColor } = theme;
  const currentRole = extractRoleFromPath(pathname);

  const [previousRole, setPreviousRole] = useState(currentRole);
  useEffect(() => setPreviousRole(currentRole), [previousRole, currentRole]);

  const didRouteChanged = previousRole !== currentRole;
  const { primaryColor: previousPrimaryColor } = generateColors(previousRole);

  const bgTransitions = useTransition(pathname, currentRole, {
    from: { backgroundColor: didRouteChanged ? previousPrimaryColor : primaryColor },
    enter: { backgroundColor: primaryColor },
    leave: { backgroundColor: primaryColor },
    config: config.wobbly
  });

  return bgTransitions.map(
    ({ item, props, key }) =>
      item && (
        <Wrapper key={key} style={props}>
          {children}
        </Wrapper>
      )
  );
}

export default withTheme(Animator);
