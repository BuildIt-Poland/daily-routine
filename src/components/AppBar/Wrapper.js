import styled from 'styled-components';

import media from '../../styles/media';
import { colorWhite, spacingMedium } from '../../styles/designTokens';

const Wrapper = styled.header`
  width: 100%;
  background: ${colorWhite};
  height: 6rem;
  display: flex;
  align-items: center;
  justify-content: center;
  padding: 0 ${spacingMedium};

  ${media.phone`
    justify-content: flex-start;
  `};
`;

export default Wrapper;
