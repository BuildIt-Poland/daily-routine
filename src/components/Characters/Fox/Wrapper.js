import styled from 'styled-components';

import media from '../../../styles/media';
import { spacingLarge } from '../../../styles/designTokens';

const Wrapper = styled.div`
  padding: 3rem 0 ${spacingLarge} 0;
  position: relative;

  ${media.phone`
    padding-bottom: 0;
  `};
`;

export default Wrapper;
