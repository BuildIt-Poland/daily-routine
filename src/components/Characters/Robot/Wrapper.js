import styled from 'styled-components';

import CharacterWrapper from '../CharacterWrapper';

const Wrapper = styled(CharacterWrapper)`
  padding-top: 0.4rem;
`;

export default Wrapper;
