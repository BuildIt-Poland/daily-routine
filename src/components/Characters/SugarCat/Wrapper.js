// Positions SugarCat at the center

import styled from 'styled-components';

import { spacingMedium } from '../../../styles/designTokens';

const Wrapper = styled.div`
  display: flex;
  justify-content: center;
  margin-top: ${spacingMedium};
`;

export default Wrapper;
