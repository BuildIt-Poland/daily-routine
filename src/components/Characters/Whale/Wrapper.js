import styled from 'styled-components';

import { spacingLarge } from '../../../styles/designTokens';

const Wrapper = styled.div`
  padding: 8rem 0 ${spacingLarge} 0;
  position: relative;
`;

export default Wrapper;
