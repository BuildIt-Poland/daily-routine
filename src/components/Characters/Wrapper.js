import styled from 'styled-components';

import { spacingLarge } from '../../styles/designTokens';

const Wrapper = styled.div`
  margin: 7.6rem 0 ${spacingLarge} 0;
  position: relative;
`;

export default Wrapper;
