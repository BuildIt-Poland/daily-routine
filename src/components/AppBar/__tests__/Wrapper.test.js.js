import React from 'react';
import { create } from 'react-test-renderer';

import Wrapper from '../Wrapper';

describe('COMPONENT - AppBar Wrapper', () => {
  it('renders correctly', () => {
    const component = create(<Wrapper />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
