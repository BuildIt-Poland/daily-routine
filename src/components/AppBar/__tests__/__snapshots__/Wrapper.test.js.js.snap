// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - AppBar Wrapper renders correctly 1`] = `
.c0 {
  width: 100%;
  background: #ffffff;
  height: 6rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 0 2rem;
}

@media (max-width:26.25em) {
  .c0 {
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
  }
}

<header
  className="c0"
/>
`;
